### DISCUSSION TITLE
Eager loading to get role names

### DISCUSSION THREAD

**Discussion by Fanreza** [spatie/laravel-permission#2130] [Q&A]
```
@foreach($user->getRoleNames() as $role)
	<label class="badge rounded-pill bg-primary">{{ $role }}</label>
@endforeach
```

i tried to figure out how to get role names without using getRoleNames function in blade, because when i saw in clockwork, there are many queries from getRoleNames, is it normal or n+1 problem

**Comment by erikn69** [comment]
Use eager loader first

**Reply by Fanreza** [reply]
is it like this?
`public function index(Request $request)

    {
        $data = User::orderBy('id','DESC');
        return view('users.index',compact('data'))->with('roles');
    }`

and how can i access it in blade view?

**Reply by erikn69** [reply]
Eager loading is for models, not for views
That code has no sense, it is working for you?

Read documentation for learn eager loading
[eloquent-relationships#eager-loading](https://laravel.com/docs/9.x/eloquent-relationships#eager-loading)
